tutorials/notebook.md

```diff
@@ -0,0 +1,569 @@
+    }
+}
+```
+
+Next, run `npm install` from the application directory to install the dependency
+to your local cache.
+
+After installing the `tupelo-client` dependency, make a new file called
+`index.js` with your favorite editor to hold all the application code. At the
+top of that file, require the Tupelo client library so we can use it later.
+
+In file `notebook/index.js`:
+```javascript
+const tupelo = require('tupelo-client');
+```
+
+## Creating a New Notebook
+Our application will organize notes in _notebooks_, and we'll store the data
+associated with a particular notebook in its own, unique chain tree. Before we
+can start saving notes, we need to register a wallet at the RPC server to store
+the keys used to access the chain tree, create a new key, and create the chain
+tree we'll use to store our notes. Let's build a `createNotebook()` function,
+step by step, to do all those things.
+
+### Connecting to the RPC Server
+The Tupelo RPC server listens for connections on a specific port of the host its
+running on after it starts. Since we did not specify the port [when we started
+the server](#running-the-tupelo-rpc-server), our server is listening on "50051",
+which is the default port. We will also need to supply a set of wallet
+credentials to establish every connection to the RPC server. The credentials are
+in an object with `walletName` and `passPhrase` keys.
+
+Let's make a `connect` function that accepts wallet credential objects and
+establishes a connection to the running development server.
+
+In file `notebook/index.js`:
+```javascript
+function connect(creds) {
+  return tupelo.connect('localhost:50051', creds);
+};
+```
+
+Next, let's start building the `createNotebook()` function by establishing a
+connection to the RPC server
+
+In file `notebook/index.js`:
+```javascript
+function createNotebook(creds) {
+  let client = connect(creds);
+}
+```
+
+### Registering a New Wallet
+In order to create and manipulate a chain tree in Tupelo, you must first create
+or control the key associated with that chain tree. The RPC server manages those
+keys in different wallets, but the client has to create the wallet and set its
+credentials first. The `register()` method of the tupelo client returns a
+promise that, when fulfilled, either contains the result of creating a new
+wallet with the client's credentials, or an error if the wallet already
+exists or there's another error during creation. Let's extend our
+`createNotebook()` function to register a new wallet.
+
+In file `notebook/index.js`:
+```javascript
+function createNotebook(creds) {
+  let client = connect(creds);
+
+  client.register()
+    .then(function(registerResult){
+      console.log("Success!")
+    }, function(err) {
+      console.log("Error registering wallet.");
+      console.log(err);
+    });
+}
+```
+
+Upon success, we just print to the console that the action succeeded. In case an
+error happened, we print the error to the console as well.
```

Comment:
Would it make sense here to explicitly have the reader run the app in its current state to ensure it's connecting to the RPC server?

Reply:
Looks like the other issue is that if you run it more than once, it fails because the wallet already exists (assuming you hardcode a value for that and don't change it each time). I wonder if there's a simple workaround for that like appending the time to the name or an idempotent version of `register` that we could call instead?

Reply:
This is another good suggestion, and I think the way to fix it is with an idempotent `register()` function that doesn't fail when the wallet already exists, but returns an unobtrusive error instead. I think that's a good thing to add on soon and should be pretty straightforward.